lib/surrounded/context/name_collision_detector.rb

```diff
@@ -0,0 +1,43 @@
+module NameCollisionDetector
+
+
+  def on_name_collision(handler)
+    case handler
+    when :nothing
+      @handler = lambda {|role, array| }
+    when :raise_exception
+      @handler = lambda {|role, array| raise Surrounded::Context::NameCollisionError.new "#{role} has name collisions with #{array}"}
+    when :warn
+      @handler = lambda {|role, array| puts "#{role} has name collisions with #{array}" if array.length > 0}
+    else
+      @handler = handler
+    end
```

Comment:
I think you get the core of the argument when you point out that warn only accepts a string. My intention was to pass as much detail to the handler as possible, in the form of the role that had the name collision, and which other role names were colliding with it. If we limit ourselves to passing only a string then that is all that the client can use which may not be terribly helpful. Furthermore, if we want to use :nothing then that must be defined somewhere in the code base too. However, I do think that your solution is cleaner and in any case it may be worthwhile using your if/else for the default case.